src/main/java/seedu/address/ui/TaskListPanel.java

```diff
@@ -0,0 +1,50 @@
+package seedu.address.ui;
+
+import java.util.logging.Logger;
+
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.assignment.Task;
+
+/**
+ * Panel containing the list of assignments.
+ */
+public class TaskListPanel extends UiPart<Region> {
+    private static final String FXML = "TaskListPanel.fxml";
+    private final Logger logger = LogsCenter.getLogger(TaskListPanel.class);
+
+    @FXML
+    private ListView<Task> taskListView;
+
+    /**
+     * Creates a {@code AssignmentListPanel} with the given {@code ObservableList}.
+     */
+    public TaskListPanel(ObservableList<Task> taskList) {
+        super(FXML);
+        taskListView.setItems(taskList);
+        taskListView.setCellFactory(listView -> new TaskListViewCell());
+    }
+
+    /**
+     * Custom {@code ListCell} that displays the graphics of a {@code Assignment} using a {@code AssignmentCard}.
+     */
+    class TaskListViewCell extends ListCell<Task> {
+
+        protected void updateItem(Task task, boolean empty) {
+            super.updateItem(task, empty);
```

Comment:
Similar to the above comment regarding JavaDocs

Reply:
amended!